src/build.ts

```diff
@@ -87,15 +87,37 @@ export class SnapcraftBuilder {
     return await fs.promises.readdir(dir)
   }
 
+  getOuputSnapCount(): number {
+    const argsArr = this.snapcraftArgs.split(' ')
+    for (const [index, element] of argsArr.entries()) {
+      if (element.includes('--build-for') || element.includes('--build-on')) {
+        let archArg = null
```

Comment:
IMO neater (so that no string optional is needed to be declared) would be: 

```typescript
      if (element.includes('--build-for') || element.includes('--build-on')) {
        let archArg = element.includes('=')
          ? element.split('=')[1]
          : argsArr[index + 1]
```